shell/lisa_shell

```diff
@@ -42,7 +45,9 @@ source "$LISA_HOME/shell/lisa_colors"
 export _DOC_LISA_HOST_ABI="Add some shell utilities to the PATH, with lower
 priority than system's one in case the user needs a different version of them"
 export LISA_HOST_ABI=${LISA_HOST_ABI:-x86_64}
-export PATH=$PATH:$LISA_HOME/shell/:$LISA_HOME/tools/$LISA_HOST_ABI:$LISA_HOME/tools
+
+_LISA_PATH="$LISA_HOME/shell/:$LISA_HOME/tools/$LISA_HOST_ABI:$LISA_HOME/tools/:$ANDROID_HOME/tools/:$ANDROID_HOME/platform-tools/"
+export PATH=$_LISA_PATH:$PATH
```

Comment:
I'm really not fond of overriding the user choices. One of the requirements for our install of ADB/fastboot was that the user's one would be used in priority (@derkling argued for it), and I think the same applies for trace-cmd.

Yes, if your trace-cmd is busted (e.g. the one shipped by most distros), you'll get a segfault. But if you have a different trace-cmd already set up on your system (maybe you're working on a fancy custom kernel and you built your own trace-cmd to make it work with your hacks, or you just need a newer trace-cmd in general), then you would expect that to be used.

It's an "ease of first time use" vs "don't mess up existing user's setup" balancing, and I heavily lean towards the second one.

Reply:
Giving priority to LISA means you will just have to delete the binary in LISA's folder if you want to use the systems' one. This is easy to do, can be supported and won't break randomly. Deleting a file that is part of a package is a great way to break your system and is definitely *not* something that any package manager even remotely supports.

> ease of first time use
It's not about first time use. It's about use, even after a 1000 times. The problem does not go away at the second use, and fixing currently requires a patch to LISA that will need to evolve as things we add to PATH change. OTH, "fixing" the priority if you don't want LISA to mess up with your system binary is straightforward: delete some LISA binaries.

To break even, it would be easy to add a config option to let the user choose which PATH order they want. Default would still be to make lisa commands work. `source init_env` is supposed to create an environment in which all the `lisa-` commands we provide work. Since LISA is mostly distro-agnostic, that means we have to ship versions known to work for some tools. The user can open a separate shell if they need their distro's trace-cmd.

Now if we were to make things "perfectly", we would provide a Debian package, a Ubuntu package, an Archlinux package and maybe Fedora. It would contain a build of trace-cmd and would explicitly conflict/replace the systems' trace-cmd. The package manager would then be happy and the user will be aware of that, and end up with only one version of the tool. We are not shipping such packages.

Reply:
I think both points are valid. But general expectations is that system's binary take a priority, and silent override will catch some people out. At the same time if a broken trace-cmd is installed (like what happened to me when I first started using Lisa), it'd confuse the hell out of the user and make them silently drop Lisa.

As a middle ground, one can prompt the user to a series of questions when they do `. init_env` so that they can choose to use a prebuilt one or use their own. Or shout loudly it was overriden and give an option to set an ENV_VARIABLE to provide a path to trace-cmd they want to use etc. I think defaulting to a stable setup is a good idea given that we make it obvious how the user can take back control if they want to. Nice out of box experience is good to help wider adoption IMHO.

Reply:
As it is `source init_env` is supposed to work in non-interactive environment, but we could provide an interactive one if we want to start asking for user input like that.

Ultimately, people will install kernelshark since it's in install_ubuntu_base.sh, which provides a trace-cmd binary as well (that's not optional, installing kernelshark on Ubuntu implies installing trace-cmd). "taking back control" by asking affect people to delete binaries installed in system-wide location and managed by the distro package manager is *not* a valid approach. If that step is necessary to get LISA working, LISA is either broken or not supporting the distro. "unsupporting" distro just because of that when we have working binaries that we already ship would make no sense.

The only real choice we have here are:
* give priority to LISA binary.
* rename LISA binaries to something like lisa-trace-cmd, so both the distro one and the LISA one can coexist in lisa-shell.
* use a shim for trace-cmd that selects the one with the highest version number.
* use a shim for binaries we ship that will look at some env var to know what binary to use.

I'll update the PR with the last proposition implemented, so people can selectively choose what binary to use for each tool we ship if necessary.